spec.html

```diff
@@ -1079,13 +1079,14 @@ <h1>RunClassFinishers ( constructor, finishers )</h1>
     </emu-clause>
 
     <emu-clause id=sec-decorate-class aoid=DecorateClass>
-      <h1>DecorateClass ( elements, decorators )</h1>
+      <h1>DecorateClass ( _elements_, _decorators_ )</h1>
       <emu-alg>
-        1. Let _newElements_, _finishers_, and _keys_ each be a new empty List.
+        1. Let _newElements_ and _finishers_ each be a new empty List.
+        1. Let _placements_ be the Record { [[StaticKeys]]: « », [[PrototypeKeys]]: « », [[OwnKeys]]: « » }.
         1. For each _element_ in _elements_, do
-          1. Append _element_.[[Key]] to _keys_.
+          1. Perform ? AddElementPlacement(_element_, _placements_, *true*).
```

Comment:
Initially I assumed that `elements` was already deduplicated from `CoalesceClassElements`, so it would be fine to reuse `AddElementPlacement` as it wouldn't introduce an error for user-supplied declarations. However, upon rereading that algorithm it seems that we don't coalesce field initializers, so there exists the possibility of user-written duplicates. Since this mechanism is only to prevent _decorators_ from adding a duplicate definition, I've modified `AddElementPlacement` to accept an optional `silent` argument to suppress errors when we initially create `placements`.

Reply:
Good call, makes sense to me.